sympy/polys/rootoftools.py

```diff
@@ -772,15 +775,94 @@ def _get_roots(cls, method, poly, radicals):
         free_names = {str(i) for i in poly.free_symbols}
         for x in chain((symbols('x'),), numbered_symbols('x')):
             if x.name not in free_names:
+                # xreplace makes alg field domain back into EX
+                # XXX: so make sure it stays same?
                 poly = poly.xreplace({d: x})
+                if dom.is_AlgebraicField:
+                    poly = Poly(poly, domain=dom)
                 break
+
+        if dom.is_QQ or dom.is_ZZ:
+            return cls._get_roots_qq(method, poly, radicals)
+            # XXX: any better way to check if real?
+            # without this check, some doctests fail (with complex terms)
+        if dom.is_AlgebraicField and all(c.is_real for c in poly.coeffs()):
+            return cls._get_roots_alg(method, poly, radicals)
+        else:
+            # XXX: not sure how to handle ZZ[x] which appears in some tests?
+            # this makes the tests pass alright but has to be a better way?
+            return cls._get_roots_qq(method, poly, radicals)
```

Comment:
One example is in integrals/tests/test_prde.py under test_param_rischDE, line 224. It seems it tries to compute the real roots of the following.
```
Poly((1 - x)*_z + 1 - x, _z, domain='ZZ[x]')
```
Without the `else` return, `None` is returned and then it throws a TypeError. 

This is reproducible (if the final `else:return` is removed) as:
```
>>> from sympy import *
>>> from sympy.abc import x,z
>>> print(Poly((1 - x)*z + 1 - x, z, domain='ZZ[x]').real_roots())
None
>>> 
```

But adding in the final `else: return cls._get_roots_qq()` fixes this. The correct answer should be that the root is -1.

Reply:
> The correct answer should be that the root is -1.

I think the correct answer should be that `real_roots` rejects this polynomial because the domain is non-numeric.

See also gh-26787

Reply:
I'm uncertain about this because if the final `else: return` is removed, three tests break, the one above and two more.
They are:
- sympy/integrals/tests/test_prde.py:224: in test_param_rischDE
- sympy/integrals/tests/test_risch.py:206: in test_recognize_log_derivative
- sympy/series/tests/test_limitseq.py:81: in test_limit_seq

The polynomials in question are respectively:
- `Poly((1 - x)*_z + 1 - x, _z, domain='ZZ[x]')`
- `Poly((2*x**4 - 8*x**3 + 8*x**2)*z**2 - 2*x**4 + 8*x**3 - 8*x**2, z, domain='ZZ[x]')`
- `Poly(288*_p + 288*EulerGamma, _p, domain='ZZ[EulerGamma]')`

I'm not sure if `real_roots()` should reject these, or if these functions should somehow be changed so that the domain is numeric, or if they should be solved another way, perhaps with `ground_roots()`?